File: backend/src/Contracts/Auth/UpdateUserProfileRequest.cs
namespace OptimaHealthcare.Contracts.Auth;

public sealed class UpdateUserProfileRequest
{
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string MobileNumber { get; set; } = string.Empty;
    public string EmailAddress { get; set; } = string.Empty;
    public string? NewPassword { get; set; }
    public string? ProfileImage { get; set; }
    public string? ImageBase64 { get; set; }
    public string? ImageFileName { get; set; }
    public string? ImageContentType { get; set; }
}

File: mobile/lib/features/auth/models/update_user_profile_request.dart
class UpdateUserProfileRequestModel {
  UpdateUserProfileRequestModel({
    required this.firstName,
    required this.lastName,
    required this.mobileNumber,
    required this.emailAddress,
    this.profileImage,
    this.imageBase64,
    this.imageFileName,
    this.imageContentType,
    this.newPassword,
  });

  final String firstName;
  final String lastName;
  final String mobileNumber;
  final String emailAddress;
  final String? profileImage;
  final String? imageBase64;
  final String? imageFileName;
  final String? imageContentType;
  final String? newPassword;

  Map<String, dynamic> toJson() {
    return {
      'firstName': firstName,
      'lastName': lastName,
      'mobileNumber': mobileNumber,
      'emailAddress': emailAddress,
      'profileImage': profileImage,
      'imageBase64': imageBase64,
      'imageFileName': imageFileName,
      'imageContentType': imageContentType,
      'newPassword': (newPassword == null || newPassword!.isEmpty)
          ? null
          : newPassword,
    };
  }
}
